feathr_project/test/test_azure_spark_e2e.py

```diff
@@ -449,9 +449,65 @@ def test_feathr_materialize_with_time_partition_pattern():
 
     # download result and just assert the returned result is not empty
     # by default, it will write to a folder appended with date
-    res_df = get_result_df(client_consumer, "avro", output_path_tpp + "/df0/daily/2020/05/20")
+    res_df = get_result_df(client_consumer, "avro", output_path_tpp + "/df0/daily/2020/05/21")
+    assert res_df.shape[0] > 0
+    
+def test_feathr_materialize_with_time_partition_pattern_hourly():
+    """
+    Test FeathrClient() using HdfsSource with 'timePartitionPattern'.
+    """   
+    test_workspace_dir = Path(
+        __file__).parent.resolve() / "test_user_workspace"
+    # os.chdir(test_workspace_dir)
+    # Create data source first
+    client_producer: FeathrClient = basic_test_setup(os.path.join(test_workspace_dir, "feathr_config.yaml"))
+
+    backfill_time = BackfillTime(start=datetime(
+        2020, 5, 20), end=datetime(2020, 5, 20), step=timedelta(days=1))
+
+    if client_producer.spark_runtime == 'databricks':
+        output_path = 'dbfs:/timePartitionPattern_hourly_sample'
```

Comment:
better to add some time signature in the output path so each job's output can be unique

Reply:
this is used as data source not the output we want to check so maybe we don't need to take extra space to save it